lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -64,21 +58,37 @@ func newForwardInterceptor(server *Server, stream Router_HtlcInterceptorServer)
 // to read from the client stream.
 // To coordinate all this and make sure it is safe for concurrent access all
 // packets are sent to the main where they are handled.
-func (r *forwardInterceptor) run() error {
+func (r *forwardInterceptor) attachStream(stream Router_HtlcInterceptorServer) error {
 	// make sure we disconnect and resolves all remaining packets if any.
 	defer r.onDisconnect()
 
-	// Register our interceptor so we receive all forwarded packets.
-	interceptableForwarder := r.server.cfg.RouterBackend.InterceptableForwarder
-	interceptableForwarder.SetInterceptor(r.onIntercept)
-	defer interceptableForwarder.SetInterceptor(nil)
+	r.stream = stream
+
+	if !r.server.cfg.RequireInterceptor {
+		// Register our interceptor so we receive all forwarded packets.
+		interceptableForwarder := r.server.cfg.RouterBackend.InterceptableForwarder
+		interceptableForwarder.SetInterceptor(r.onIntercept)
+		defer interceptableForwarder.SetInterceptor(nil)
+	}
 
 	// start a go routine that reads client resolutions.
 	errChan := make(chan error)
 	resolutionRequests := make(chan *ForwardHtlcInterceptResponse)
 	r.wg.Add(1)
 	go r.readClientResponses(resolutionRequests, errChan)
 
+	// If we have any previously held forwards, re-send them in case they
+	// didn't reach the interceptor the first time.
+	for _, held := range r.holdForwards {
+		log.Tracef("re-sending intercepted packet to client after reconnect %v", held.Packet())
+
+		if err := r.sendRequestForHtlc(held.Packet()); err != nil {
```

Comment:
A test case simulating a lost connection here and asserting that held htlcs are replayed upon reconnect would be nice.